Sort assets by modified date

MrFlashAccount opened the pull request:
### Pull Request Description

Closes: https://github.com/enso-org/cloud-v2/issues/1611, https://github.com/enso-org/cloud-v2/issues/1602

This PR refactors the `EditableSpan` component and improve ergonomics of editing file names.

Now:
1. It displays loading state
2. It shows error if something goes wrong
3. It validates the name
4. Changed behavior of comparing asset names: now it's case insensitive and compare across all assets

Also, this PR changes the default order of assets in Cloud backend to match the order with the Local one

This PR fixes the tooltip when a project being executed by another user


https://github.com/user-attachments/assets/83a0cfd6-b090-497c-8171-729c372b00c8


https://github.com/user-attachments/assets/fcf12082-04a3-4fd3-821e-20519343913a


<!--
- Please describe the nature of your PR here, as well as the motivation for it.
- If it fixes an open issue, please mention that issue number here.
-->

### Important Notes

<!--
- Mention important elements of the design.
- Mention any notable changes to APIs.
-->

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.
- [ ] If meaningful changes were made to logic or tests affecting Enso Cloud integration in the libraries, 
      or the Snowflake database integration, a run of the [Extra Tests](https://github.com/enso-org/enso/actions/workflows/extra-nightly-tests.yml) has been scheduled.
  - If applicable, it is suggested to paste a link to a successful run of the Extra Tests.

MrFlashAccount committed:
Sort assets by modified date

MrFlashAccount committed:
Add error state to Editable span

MrFlashAccount committed:
Check for uniqness and display error accordingly

MrFlashAccount committed:
Fix tiny issues

MrFlashAccount committed:
Fix `used by another dev behavior`

MrFlashAccount committed:
Merge branch 'develop' into wip/sergeigarin/fix-concurrent-access

# Conflicts:
#	app/gui/integration-test/dashboard/actions/api.ts
#	app/gui/integration-test/dashboard/actions/index.ts
#	app/gui/integration-test/dashboard/assetsTableFeatures.spec.ts
#	app/gui/src/dashboard/layouts/AssetsTable.tsx

MrFlashAccount committed:
Merge branch 'develop' into wip/sergeigarin/fix-concurrent-access

MrFlashAccount committed:
FIxes

MrFlashAccount committed:
Remove unused import

MrFlashAccount commented on app/gui/src/dashboard/App.tsx:
```diff
@@ -519,40 +518,38 @@ function AppRouter(props: AppRouterProps) {
   )
 
   return (
-    <FeatureFlagsProvider>
```
Here I only removed the `FeatureFlagsProvider`

4e6 reviewed (approved):
The tools part looks good to me

MrFlashAccount committed:
FIxes x2

MrFlashAccount committed:
Add border

MrFlashAccount committed:
Improve animations

MrFlashAccount committed:
Fix animation

MrFlashAccount committed:
Fix overflow

MrFlashAccount committed:
Fix eslint

MrFlashAccount committed:
Fix lint

MrFlashAccount committed:
Fix lint issues

MrFlashAccount committed:
Fix docs for underaly element

MrFlashAccount commented on app/gui/src/dashboard/providers/FeatureFlagsProvider.tsx:
```diff
@@ -4,55 +4,82 @@
 
-const flagsStore = createStore<FeatureFlags>()(
+const flagsStore = createStore<FeatureFlagsStore>()(
   persist(
```
@somebody1234 I found that the `zustand` has native support for persisting state, wdyt?

MrFlashAccount commented on app/gui/src/dashboard/providers/TextProvider.tsx:
```diff
@@ -70,18 +68,12 @@ export default function TextProvider(props: TextProviderProps) {
+
+  const localizedText = text.getDictionary(language)
 
   const getText = React.useCallback<GetText>(
```
This function is pulled off the provider, so we can now use it outside of React.

PabloBuchu commented:
1. Loader after rename sometimes properly waits till title is updated sometimes it sets the name back to old one and refetch updates to new value

https://github.com/user-attachments/assets/11f9db35-7008-4dcd-a725-ad4ea95f0d1a

2. Here you can see the given project is in `Opened` state but it shows as `Closed`
<img width="1728" alt="Screenshot 2024-12-19 at 09 33 40" src="https://github.com/user-attachments/assets/e7000977-f315-42cb-bb06-478153130444" />

MrFlashAccount commented:
> Loader after rename sometimes properly waits till title is updated sometimes it sets the name back to old one and refetch updates to new value


It's the backend issue, we discussed it, it requires a fix on the backend

> Here you can see the given project is in Opened state but it shows as Closed

Oh intersting. Will take a look. Guess there are some out of sync state between the `listDirectories` and the `getProjectDetails`

@PabloBuchu is it possible though?

MrFlashAccount committed:
Merge branch 'develop' into wip/sergeigarin/fix-concurrent-access

# Conflicts:
#	app/gui/env.d.ts

JaroslavTulach committed:
Promote broken values instead of ignoring them (#11777)

Partially fixes #5430 by propagating `DataflowError`s found during statement execution out of the method.

# Important Notes
This change [may affect behavior](https://github.com/enso-org/enso/pull/11673/files#r1871128327) of existing methods that ignore `DataflowError` as [discussed here](https://github.com/enso-org/enso/pull/11673/files#r1871128327).

jdunkerley committed:
Missed one :( (#11918)

MrFlashAccount committed:
Try fix opened projects

MrFlashAccount committed:
Fix tests

MrFlashAccount committed:
remove console logs

MrFlashAccount commented:
@PabloBuchu addressed the second issue. Please take a look once you have time 😊

MrFlashAccount committed:
Merge branch 'develop' into wip/sergeigarin/fix-concurrent-access

MrFlashAccount committed:
Merge branch 'develop' into wip/sergeigarin/fix-concurrent-access

# Conflicts:
#	app/gui/src/dashboard/components/AriaComponents/Form/components/FormError.tsx
